mongo/mgo_job_store.go

```diff
@@ -0,0 +1,198 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+	dpMongodb "github.com/ONSdigital/dp-mongodb"
+	dpMongoLock "github.com/ONSdigital/dp-mongodb/dplock"
+	dpMongoHealth "github.com/ONSdigital/dp-mongodb/health"
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+)
+
+// MgoJobStore defines the required methods from MongoDB
+type MgoJobStore interface {
```

Comment:
Similarly to the filename, this is a `JobStore` in the `mongo` so the Mgo prefix is not needed. Also there's no need to specify the library we are using as part of the name. If we choose to use something other than `mgo` in future, we wouldn't want to have to change this package's interface or have it being misleading.
```suggestion
type JobStorer interface {
```

Additionally, the interface belongs in the package that is using it in go. This is different from Java where the interface is defined by the package that owns the implementation in many cases. This is due to the way interfaces are defined in go. So for instance…
In Java: A class might say 'I am xyz and I implement interface abc'
In Go: A package might say 'I need something that satisfies the interface I am defining called abc'. The other type doesn't need to explicitly define that it satisfies that interface it just has to have all the required functions.

For this reason I'd move the interface to where it's used. Also it's OK for different packages to define different interfaces for only the things they need.